tests/Microsoft.Data.Analysis.IO.Tests/Microsoft.Data.Analysis.IO.Tests/DataFrame.IOTests.cs

```diff
@@ -0,0 +1,638 @@
+            VerifyColumnTypes(df);
+        }
+
+        [Fact]
+        public void TestReadCsvNoHeader()
+        {
+            string data = @"CMT,1,1,1271,3.8,CRD,17.5
+CMT,1,1,474,1.5,CRD,8
+CMT,1,1,637,1.4,CRD,8.5
+CMT,1,1,181,0.6,CSH,4.5";
+
+            Stream GetStream(string streamData)
+            {
+                return new MemoryStream(Encoding.Default.GetBytes(streamData));
+            }
+            DataFrame df = DataFrameIO.LoadCsv(() => GetStream(data), SetTextFieldParserProperties, header: false);
+            Assert.Equal(4, df.Rows.Count);
+            Assert.Equal(7, df.Columns.Count);
+            Assert.Equal("CMT", df.Columns["Column0"][3]);
+            VerifyColumnTypes(df);
+
+            DataFrame reducedRows = DataFrameIO.LoadCsv(() => GetStream(data), SetTextFieldParserProperties, header: false, numberOfRowsToRead: 3);
+            Assert.Equal(3, reducedRows.Rows.Count);
+            Assert.Equal(7, reducedRows.Columns.Count);
+            Assert.Equal("CMT", reducedRows.Columns["Column0"][2]);
+            VerifyColumnTypes(df);
+        }
+
+        void VerifyDataFrameWithNamedColumnsAndDataTypes(DataFrame df, bool verifyColumnDataType, bool verifyNames)
+        {
+            Assert.Equal(4, df.Rows.Count);
+            Assert.Equal(7, df.Columns.Count);
+
+            if (verifyColumnDataType)
+            {
+                Assert.True(typeof(string) == df.Columns[0].DataType);
+                Assert.True(typeof(short) == df.Columns[1].DataType);
+                Assert.True(typeof(int) == df.Columns[2].DataType);
+                Assert.True(typeof(long) == df.Columns[3].DataType);
+                Assert.True(typeof(float) == df.Columns[4].DataType);
+                Assert.True(typeof(string) == df.Columns[5].DataType);
+                Assert.True(typeof(double) == df.Columns[6].DataType);
+            }
+
+            if (verifyNames)
+            {
+                Assert.Equal("vendor_id", df.Columns[0].Name);
+                Assert.Equal("rate_code", df.Columns[1].Name);
+                Assert.Equal("passenger_count", df.Columns[2].Name);
+                Assert.Equal("trip_time_in_secs", df.Columns[3].Name);
+                Assert.Equal("trip_distance", df.Columns[4].Name);
+                Assert.Equal("payment_type", df.Columns[5].Name);
+                Assert.Equal("fare_amount", df.Columns[6].Name);
+            }
+
+            VerifyColumnTypes(df);
+
+            foreach (var column in df.Columns)
+            {
+                Assert.Equal(0, column.NullCount);
+            }
+        }
+
+        [Theory]
+        [InlineData(true, 0)]
+        [InlineData(false, 0)]
+        [InlineData(true, 10)]
+        [InlineData(false, 10)]
+        public void TestReadCsvWithTypesAndGuessRows(bool header, int guessRows)
+        {
+            /* Tests this matrix
+             * 
+                header	GuessRows	DataTypes	
+                True	0	        NotNull	    
```

Comment:
Nit: can this be formatted (as intended)?

Reply:
It is actually. I don't know why GH doesn't render it correctly.